api/v1alpha1/clusterconfig_types.go

```diff
@@ -175,6 +180,88 @@ func (ExtraAPIServerCertSANs) VariableSchema() clusterv1.VariableSchema {
 	}
 }
 
+type ImageRegistries struct {
+	// +optional
+	ImageRegistryCredentials ImageRegistryCredentials `json:"credentials,omitempty"`
+}
+
+func (ImageRegistries) VariableSchema() clusterv1.VariableSchema {
+	return clusterv1.VariableSchema{
+		OpenAPIV3Schema: clusterv1.JSONSchemaProps{
+			Description: "Configuration for image registries.",
+			Type:        "object",
+			Properties: map[string]clusterv1.JSONSchemaProps{
+				"credentials": imageRegistryCredentialsSchema,
+			},
+		},
+	}
+}
+
+var (
+	imageRegistryCredentialsSchema = clusterv1.JSONSchemaProps{
+		Type:        "array",
+		UniqueItems: true,
+		Items:       &imageRegistryCredentialsResourceSchema,
+	}
+
+	imageRegistryCredentialsResourceSchema = clusterv1.JSONSchemaProps{
+		Description: "Image registry credentials to set up on all Nodes in the cluster. " +
+			"Enabling this will the Kubelets with https://kubernetes.io/docs/tasks/administer-cluster/kubelet-credential-provider/.",
+		Type: "object",
+		Properties: map[string]clusterv1.JSONSchemaProps{
+			"url": {
+				Description: "Registry URL.",
+				Type:        "string",
+			},
+			"secretRef": {
+				Description: "The Secret containing the registry credentials. " +
+					"The Secret should have keys 'username' and 'password'. " +
+					"This credentials Secret is not required for some registries, e.g. ECR.",
+				Type: "object",
+				Properties: map[string]clusterv1.JSONSchemaProps{
+					"name": {
+						Description: "The name of the Secret containing the registry credentials.",
+						Type:        "string",
+					},
+					"namespace": {
+						Description: "The namespace of the Secret containing the registry credentials. " +
+							"Defaults to the namespace of the KubeadmControlPlaneTemplate and KubeadmConfigTemplate" +
+							" that reference this variable.",
+						Type: "string",
+					},
```

Comment:
Let's make this a `ObjectReference` schema so we can reuse it in other places when required.